Assets/Plugins/Clean/Editor/Upgrade.cs

```diff
@@ -0,0 +1,92 @@
+using System;
+using System.Collections.Generic;
+using TMPro;
+using UnityEngine;
+using UnityEngine.SceneManagement;
+using UnityEngine.UI;
+using Object = UnityEngine.Object;
+
+namespace Plugins.Clean.Editor
+{
+    public class Upgrade
+    {
+        private List<KeyValuePair<GameObject, ComponentProperties.Text>> textCache =
+            new List<KeyValuePair<GameObject, ComponentProperties.Text>>();
```

Comment:
It might be better readable if you just added the GameObject reference to the ComponentProperties.Text/TextMesh/DropDown/InputField. This way you can just have a list of those and the GameObject reference is already in there.